### DISCUSSION TITLE
AGH not working openWRT

### DISCUSSION THREAD

**Discussion by Eyal-Elbaz** [AdguardTeam/AdGuardHome#3386] [Q&A]
### Prerequisites

- [x] I checked the documentation and found no answer*
- [x] I checked to make sure that this issue has not already been filed*
- [x] I am running the latest version

*no answer that's worked
### Issue Details

* **Version of AdGuard Home server:**
  * v0.106.3
* **How did you install AdGuard Home:**
  * https://forum.openwrt.org/t/howto-running-adguard-home-on-openwrt/51678
* **How did you setup DNS configuration:**
  * Router
* **If it's a router or IoT, please write device model:**
  * wrt32x
* **Operating system and version:**
  * openwrt

### Expected Behavior
 * non-error
 * download lists
### Actual Behavior
 > getting error in startup
 > blocklists not working
### Screenshots
![img1](https://i.imgur.com/Pyx86Mp.png)
![img2](https://i.imgur.com/S6iJke7.png)

**Comment by ainar-g** [comment]
You seem to have certificate issues on your system.  If you have `curl` or `wget` installed, does `curl 'https://dns.google/'` or `wget 'https://dns.google/'` work?

**Comment by mercyground** [comment]
i'll bet he doesnt have the either of the ca-certificates or ca-bundle installed.

**Comment by Eyal-Elbaz** [comment]
@ainar-g  I need to run it with `--no-check-certificate`

**Comment by mercyground** [comment]
> @ainar-g I need to run it with `--no-check-certificate`

 opkg install ca-certificates ca-bundle

then see if it errors. Its failing because it cannot authenticate googles Certificate.

**Comment by Eyal-Elbaz** [comment]
@mercyground  I ran `opkg install ca-certificates ca-bundle` and rebooted the router but still both the errors appears

**Comment by mercyground** [comment]
what happens when you try curl 'https://dns.google/' or wget 'https://dns.google/' like ainar-g asked? if the certificates are in place it should work. Otherwise you have another issue to resolve.


```
wget 'https://dns.google/'
Downloading 'https://dns.google/'
Connecting to 8.8.8.8:443
Writing to 'index.html'

Download completed (1383 bytes)
```

**Comment by Eyal-Elbaz** [comment]
```
root@OpenWrt:~# wget 'https://dns.google/'
--2021-07-20 08:23:27--  https://dns.google/
Resolving dns.google... 2001:4860:4860::8888, 2001:4860:4860::8844, 8.8.4.4, ...
Connecting to dns.google|2001:4860:4860::8888|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: unspecified [text/html]
Saving to: 'index.html'

index.html              [ <=>                ]   1.35K  --.-KB/s    in 0s

2021-07-20 08:23:27 (18.2 MB/s) - 'index.html' saved [1383]
```

**Comment by SaifAqqad** [comment]
Installing `ca-certificates ca-bundle libustream-mbedtls libustream-openssl` then rebooting fixed it for me
